explainability-core/src/main/java/org/kie/trustyai/explainability/model/DataframeMetadata.java

```diff
@@ -0,0 +1,193 @@
+package org.kie.trustyai.explainability.model;
+
+import org.kie.trustyai.explainability.model.domain.EmptyFeatureDomain;
+import org.kie.trustyai.explainability.model.domain.FeatureDomain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class DataframeMetadata {
+    private final List<String> names;
+    private List<String> nameAliases;
+    private List<String> cachedColumnNames;
```

Comment:
Assuming this is read more frequently than written into, perhaps change it to a `CopyOnWriteArrayList`